### DISCUSSION TITLE
Typescript: How remove error "TS-2604"?

### DISCUSSION THREAD

**Discussion by trustyoo86** [vercel/next.js#17752] [Help] [answered]
Hello,
I started nextjs with typescript.

During develop nextjs app with typescript, I found TS-2602 error in `_app.tsx`.

Here is source code

```typescript
import React from 'react';
import '../styles/globals.css';

interface IProps {
  Component: React.Component[] | React.Component;
  pageProps: any;
}

function MyApp<T>({ Component, pageProps }: IProps) {
  return <Component {...pageProps} />;
}

export default MyApp;
```

Found error code is `return <Component {...pageProps} />` error message is "JSX element type 'Component' does not have any construct or call signatures."

How I remove this error in `_app.tsx`?

**Comment by jamesmosier** [comment, marked as answer]
You can change your interface to:

```js
interface IProps {
  Component: typeof React.Component;
  pageProps: any;
}
```

or

```js
interface IProps {
  Component: React.ComponentType;
  pageProps: any;
}
```

or even better would be to use `import type { AppProps } from 'next/app';` [from Next](https://nextjs.org/docs/basic-features/typescript#custom-app):

```js
function MyApp({ Component, pageProps }: AppProps): JSX.Element {
  return <Component {...pageProps} />;
}
```
